Artur Gagin: hw1

lastdesire opened the pull request.

lastdesire committed:
Upgrade main.c

lastdesire committed:
Update main.c

lastdesire committed:
Update main.c

lastdesire committed:
Update main.c

lastdesire committed:
Update main.c

lastdesire committed:
Upgrade code readability

lastdesire committed:
Upgrade code readability

lastdesire committed:
Upgrade code readability

lastdesire committed:
Upgrade code readability

lastdesire committed:
Upgrade code readability

lastdesire committed:
Upgrade code readability

lastdesire committed:
Fix bug with RETCODE

AntonKozlov commented on main.c:
```diff
@@ -1,17 +1,107 @@
+	char* name;
+	FunctionCallback function;
+} commands_list[] = {
+	{"echo",    (FunctionCallback) &echo},
+        {"retcode", (FunctionCallback) &retcode}
```
:+1:

lastdesire replied:
=)))

lastdesire replied:
^-^

AntonKozlov commented on main.c:
```diff
@@ -1,17 +1,107 @@
 	}
-	return argc - 1;
+	
+	RETCODE = argc - 1;
```
Придется писать в каждой команде?

lastdesire replied:
Пофиксил, теперь код возврата и есть RETCODE.

AntonKozlov commented on main.c:
```diff
@@ -1,17 +1,107 @@
+	char* input_delimit = "; \n";
+	if (is_parse_on_commands)
+	{
+		input_delimit = ";;;";
```
https://github.com/AntonKozlov/os226-2022/blob/master/test/batch/1.in#L8 -- здесь могло быть произвольное количество `;`

lastdesire replied:
Фиксед, однако скажите, что Вы думаете по поводу комментариев https://github.com/lastdesire/os226-2022/blob/master/main.c#L89
У меня Сегфолт на тестах не выкидывает, однако нашел ошибку (возвращаемый тип FunctionCallback был не тот). Вопрос такой: проблема кроется в том, что я написал в комментариях кода, либо там все вполне нормально и сегфолт я пофиксил / он "спрятан" в другом месте?

lastdesire replied:
Исправлено

AntonKozlov commented on main.c:
```diff
@@ -1,17 +1,107 @@
+
+//------CONST------//
+#define MAX_INPUT_SIZE 1024
+#define MAX_INPUT_ARGUMENTS 512
```
Наверно (MAX_INP_SIZE / 2) -- для читаемости

lastdesire replied:
Сделал.

AntonKozlov reviewed (commented):
Сегфолт :(

lastdesire committed:
Update mainc

1) The situation with RETCODE has been fixed: now we have to write retcode only in a command that has a special return code (for example, echo). In another case, the return code is 0.
2) Fixed a parsing error with ";". Everything should be fine now.
3) (MAX_INP_SIZE / 2) -- for reading.

lastdesire committed:
Update main.c

1) Fixed the RETCODE: now functions return int, and we change RETCODE.
2) Fixed a bug with the typedef FunctionCallBack (type changed).

lastdesire committed:
Update main.c

Fixed bug with RETCODE

lastdesire committed:
Update main.c

Fixed comment

lastdesire committed:
Update main.c

Deleted comments (I figured it out).
